.circleci/config.yml

```diff
@@ -1,52 +1,28 @@
 version: 2.1
+
 orbs:
-  ruby: circleci/ruby@0.1.2
+  ruby: circleci/ruby@1.2
 
 jobs:
-  build_2-7-2:
-    docker:
-      - image: circleci/ruby:2.7.2
-    executor: ruby/default
-    steps:
-      - checkout
-      - run:
-          name: Which bundler?
-          command: bundle -v
-      - ruby/bundle-install
-      - run:
-          name: Build CI
-          command: bundle exec rake ci
-  build_2-6-6:
+  test:
+    parameters:
+      ruby-version:
+        type: string
     docker:
-      - image: circleci/ruby:2.6.6
-    executor: ruby/default
+      - image: cimg/ruby:<< parameters.ruby-version >>
     steps:
       - checkout
-      - run:
-          name: Which bundler?
-          command: bundle -v
-      - ruby/bundle-install
-      - run:
-          name: Build CI
-          command: bundle exec rake ci
-  build_2-5-8:
-    docker:
-      - image: circleci/ruby:2.5.8
-    executor: ruby/default
-    steps:
-      - checkout
-      - run:
-          name: Which bundler?
-          command: bundle -v
-      - ruby/bundle-install
+      - ruby/install-deps:
+          bundler-version: "1.17.2"
+          with-cache: false
       - run:
           name: Build CI
           command: bundle exec rake ci
 
-# Orchestrate our job run sequence
 workflows:
   build_and_test:
     jobs:
-      - build_2-7-2
-      - build_2-6-6
-      - build_2-5-8
+      - test:
+          matrix:
+            parameters:
+              ruby-version: ["3.0", "2.7"]
```

Comment:
The oldest maintained ruby version is 2.6 (see https://www.ruby-lang.org/en/news/2021/11/24/ruby-2-6-9-released/), which is still getting security updates until March 2022.  I recommend keeping that version in your build matrix until it's EOL.  Also, you may want to add `3.1` as it was released recently.